llama_index/node_parser/relational/base_element.py

```diff
@@ -131,11 +135,19 @@ def extract_table_summaries(self, elements: List[Element]) -> None:
         llm = cast(LLM, llm)
 
         service_context = ServiceContext.from_defaults(llm=llm, embed_model=None)
-        for element in tqdm(elements):
+        for idx, element in tqdm(enumerate(elements)):
             if element.type != "table":
                 continue
+            table_context = str(element.element)
+            if idx > 0 and "table" in str(elements[idx - 1].element).lower():
```

Comment:
What is this checking for? Since we already checked for the element type above?

Reply:
get table caption from surrounding context.
Table real title is usually before/after the table

Reply:
Hmm... this feels very error prone 😅 This may cause many false positives/false negatives no?

Reply:
we let LLM pydantic program to detect. basically enrich the table context.
if no table real title detected, it just leave empty